test/e2e/framework/volume/fixtures.go

```diff
@@ -488,6 +488,17 @@ func runVolumeTesterPod(client clientset.Interface, config TestConfig, podSuffix
 
 	for i, test := range tests {
 		volumeName := fmt.Sprintf("%s-%s-%d", config.Prefix, "volume", i)
+
+		// We need to make the container privileged when SELinux is enabled on the
+		// host,  so the test can write data to a location like /tmp. Also, due to
+		// the Docker bug below, it's not currently possible to map a device with
+		// a privileged container, so we don't go privileged for block volumes.
```

Comment:
Does mean for block volume, we cannot set SELinux enabled?

Reply:
Actually, for block volume, we cannot make the *container* privileged. Specially if the container runtime is docker (due to the bug below).

Reply:
Is there another directory we can safely write to that does not require privileged?

Reply:
> Is there another directory we can safely write to that does not require privileged?

Not that I'm aware of. 

By default, containers can only RW to files/directories labeled `svirt_sandbox_file_t`. So if a process breaks out of the container, it won't be able to change the host FS.

We *could* add the label above to a pre-defined directory, say */tmp/k8s-e2e*, and change the test to read/write data there, rather than the default */tmp*. See my comment https://github.com/kubernetes/kubernetes/pull/83727#discussion_r334856930. I've tried this and it worked.

Reply:
The downside of the approach above is that we need to change the test bootstrap to do `chcon -R -t svirt_sandbox_file_t /tmp/k8s-e2e` in the host before running the test.

I don't know if we are able to do that, however, this is IMO the right approach to solve our SELinux issues in e2e tests.

Reply:
I would prefer we look into this approach of prepping a special test directory first if the environment uses SELinux. That way we can remove the privileged setting on many tests (and maybe be able to add them to the conformance suite).  But this is sufficient for now.

Reply:
Just to clarify: we can remove the privileged setting if we prep a special test directory on SELinux environments. All containers would run as *unprivileged*.

FYI, I created the issue https://github.com/kubernetes/kubernetes/issues/84585 to track this.